### PR TITLE
[4.1] Private messages reply

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #29548 .

### Summary of Changes
If a user has access to com_messages but not to com_users then they can not send a message but they should still be able to read a message

### Testing Instructions
Create a manager user and then give them full access to com_messages

### Actual result BEFORE applying this Pull Request
New button and reply button are present but you can not send a message

image

image

### Expected result AFTER applying this Pull Request
New button and reply button are not displayed

image

image

### CODE REVIEW

**Comment by Quy** [comment]
These should not be displayed.

**Comment by brianteeman** [comment]
@quy both of those should be done now - i think

File: administrator/components/com_messages/tmpl/messages/emptystate.php [hunk 1]
```diff
@@ -19,7 +19,9 @@
 	'icon'       => 'icon-envelope inbox',
 ];
 
-if (Factory::getApplication()->getIdentity()->authorise('core.create', 'com_messages'))
+if (Factory::getApplication()->getIdentity()->authorise('core.create', 'com_messages')
+	&& (Factory::getApplication()->getIdentity()->authorise('core.manage', 'com_users')))
+
 {
```
Comment (Quy) [inline administrator/components/com_messages/tmpl/messages/emptystate.php]:
```suggestion
if (Factory::getApplication()->getIdentity()->authorise('core.create', 'com_messages')
	&& Factory::getApplication()->getIdentity()->authorise('core.manage', 'com_users'))
{
```

File: administrator/modules/mod_menu/src/Menu/CssMenu.php [hunk 2]
```diff
@@ -433,6 +433,11 @@ protected function preprocess($parent)
 					continue;
 				}
 			}
+			elseif (($item->link === 'index.php?option=com_messages&view=messages' && !$user->authorise('core.manage', 'com_users')))
```
Comment (Quy) [inline administrator/modules/mod_menu/src/Menu/CssMenu.php]:
```suggestion
			elseif ($item->link === 'index.php?option=com_messages&view=messages' && !$user->authorise('core.manage', 'com_users'))
```

**Comment by Quy** [comment]
Looks good. It is still possible to go directly to it in the browser's address bar `administrator/index.php?option=com_messages&view=message&layout=edit`. Don't know if it is worth worrying about it.

**Comment by brianteeman** [comment]
Its not possible to prevent that in this case

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on dc3b9b784d7c2e3dc60b21dfef11e05eab7f3903<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37134">issues.joomla.org/tracker/joomla-cms/37134</a>.</sub>

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on dc3b9b784d7c2e3dc60b21dfef11e05eab7f3903<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37134">issues.joomla.org/tracker/joomla-cms/37134</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37134">issues.joomla.org/tracker/joomla-cms/37134</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**